mlir-sycl/test/Conversion/SYCLToGPU/grid-ops.mlir

```diff
@@ -0,0 +1,303 @@
+// RUN: sycl-mlir-opt -convert-sycl-to-gpu %s -o - | FileCheck %s
+
+!sycl_id_1_ = !sycl.id<[1], (!sycl.array<[1], (memref<1xi64, 4>)>)>
+!sycl_id_2_ = !sycl.id<[2], (!sycl.array<[2], (memref<2xi64, 4>)>)>
+!sycl_id_3_ = !sycl.id<[3], (!sycl.array<[3], (memref<3xi64, 4>)>)>
+!sycl_range_1_ = !sycl.range<[1], (!sycl.array<[1], (memref<1xi64, 4>)>)>
+!sycl_range_2_ = !sycl.range<[2], (!sycl.array<[2], (memref<2xi64, 4>)>)>
+!sycl_range_3_ = !sycl.range<[3], (!sycl.array<[3], (memref<3xi64, 4>)>)>
+
+// CHECK-LABEL:   func.func @test_num_work_items() -> !sycl_range_1_ {
+// CHECK-NEXT:      %[[VAL_0:.*]] = memref.alloca() : memref<1x!sycl_range_1_>
+// CHECK-NEXT:      %[[VAL_1:.*]] = arith.constant 0 : index
+// CHECK-NEXT:      %[[VAL_2:.*]] = memref.load %[[VAL_0]]{{\[}}%[[VAL_1]]] : memref<1x!sycl_range_1_>
+// CHECK-NEXT:      %[[VAL_3:.*]] = arith.constant 0 : i32
+// CHECK-NEXT:      %[[VAL_4:.*]] = gpu.grid_dim  x
+// CHECK-NEXT:      %[[VAL_5:.*]] = arith.index_cast %[[VAL_4]] : index to i64
+// CHECK-NEXT:      %[[VAL_6:.*]] = "sycl.range.get"(%[[VAL_2]], %[[VAL_3]]) {ArgumentTypes = [memref<1x!sycl_range_1_, 4>, i32], FunctionName = @"operator[]", MangledFunctionName = @"operator[]", TypeName = @range} : (!sycl_range_1_, i32) -> memref<1xi64, 4>
```

Comment:
Why the mangled function name is not mangled?

Reply:
It's a dummy name. When I implement the new lowering for these operations, we won't need the mangled function name and that argument will be dropped. That's why I say we cannot lower that (correctly)